docs/sources/operators-guide/configure/setting-helm-ha-deduplication-consul/index.md

```diff
@@ -0,0 +1,125 @@
+
+You also have to configure HA for Prometheus or Grafana Agent. Last, you need a KV store. In this guide you will
+use Consul. You will be guided on the setup if you haven't had one.
+
+## Configure Prometheus or Grafana Agent to send HA external labels
+
+Configure Prometheus or Grafana Agent HA setup by setting label called `cluster` and `__replica__`. These two labels
+name are default labels for HA setup in Grafana Mimir. If you want to change the labels, make sure to change it in Mimir
+too to make Grafana Mimir and Prometheus/Grafana Agent config matches with each other, otherwise HA deduplication
+wouldn't work. `cluster` label value must be same across replica belong to the same cluster. `__replica__` must be
+unique across different replica in the cluster.
+
+```yaml
+global:
+  scrape_interval: 1m
+  scrape_timeout: 10s
+  evaluation_interval: 1m
+  external_labels:
+    __replica__: replica-1
+    cluster: my-prometheus
+```
+
+Reload or restart Prometheus or Grafana Agent after updating the configuration.
+
+## Install Consul using Helm
+
+Install Consul in Kubernetes using
+[Consul helm chart](https://github.com/hashicorp/consul-k8s/tree/main/charts/consul). Follow the documentation to get
+the chart and installing a Consul release. Put a note on the Consul endpoint, because you will need it for Mimir
+configuration.
+
+## Setup Mimir HA config
+
+You can set Mimir HA deduplication configuration in global level or tenant level.
+
+### HA Deduplication Global
+
+You need a Mimir setup installed by Helm. Add the following configuration to your `values.yaml` to configure HA
+deduplication globally.
+
+```yaml
+# other configurations above
+mimir:
+  structuredConfig:
+    limits:
+      accept_ha_samples: true
+      # The following two configs must match with external_labels config in Prometheus
+      # The config values below are the default and can be removed if you don't want to override to a new value
+      ha_cluster_label: cluster
+      ha_replica_label: __replica__
+    distributor:
+      ha_tracker:
+        enable_ha_tracker: true
+        kvstore:
+          store: consul
+          consul:
+            host: <consul-endpoint> # example: http://mimir-host:8500
+```
+
+Upgrade the Mimir's helm release using the above configuration.
+
+```bash
+ helm -n <mimir-namespace> upgrade --install mimir grafana/mimir-distributed -f values.yaml
+```
+
+### HA Deduplication per tenant
+
+Add the following configuration below the `values.yaml` file to configure HA deduplication per-tenant
+
+```yaml
+runtimeConfig:
+  overrides:
+    <tenant-id>: # put real tenant ID here
+      accept_ha_samples: true
+      ha_cluster_label: cluster
+      ha_replica_label: __replica__
+```
+
+Upgrade the Mimir's helm release using the above configuration.
+
+```bash
+ helm -n <mimir-namespace> upgrade --install mimir grafana/mimir-distributed -f values.yaml
+```
+
+## Verifying deduplication
```

Comment:
> the distributor also exposes some metrics around this...
I will mention the metrics info in this section. 

> on another topic - what do you think about moving this section to the other article - "Configuring Grafana Mimir high-availability deduplication"?

I am fine to move the section to the "Configuring Grafana Mimir high-availability deduplication". But maybe we want to hear first what @krajorama or @osg-grafana think.

Reply:
Mimir does not have self monitoring like GEM so that won't make sense there in itself.
And the Mimir documentation tries to be deployment agnostic so talking about port forward seems out of scope.
Given that I'd keep these here.